flags.go

```diff
@@ -341,6 +344,82 @@ type flagStorage struct {
 	DebugMutex      bool
 }
 
+const PARENT_PROCESS_EXECUTION_DIR_KEY = "MD5Hash38cda02df9fcf2b1bba019000cce0eb6"
+
+// returns the same filepath in case of absolute path
+// returns the resolved path for relative to current dir or home dir
+func getResolvedPath(filePath string) (resolvedPath string, err error) {
+	if path.IsAbs(filePath) {
+		return filePath, nil
+	}
+
+	if strings.HasPrefix(filePath, "~") {
+		homeDir, err1 := os.UserHomeDir()
+		if err1 != nil {
+			return "", fmt.Errorf("get home dir: %w", err1)
+		}
+		return filepath.Join(homeDir, filePath[2:]), nil
+	} else {
+		return filepath.Abs(filePath)
+	}
+}
+
+// It smartly resolves the path of the required flag
+// For parent process: it only resolves the path with respect to home folder.
+// For child process (when --foreground flag is disabled) - it resolves
+// the path relative to both current directory and home directory
+func resolvePathForTheRequiredFlagInContext(c *cli.Context) (err error) {
+	parentProcessExecutionDir, ok := os.LookupEnv(PARENT_PROCESS_EXECUTION_DIR_KEY)
+	if !ok { // means parent process, hence set the parentProcessExecutionDir empty.
+		parentProcessExecutionDir = ""
+	}
+
+	err = resolvePathForTheFlagInContext("log-file", parentProcessExecutionDir, c)
+	if err != nil {
+		return fmt.Errorf("resolving for log-file: [%w]", err)
+	}
+
+	err = resolvePathForTheFlagInContext("key-file", parentProcessExecutionDir, c)
+	if err != nil {
+		return fmt.Errorf("resolving for key-file: [%w]", err)
+	}
+
+	return nil
+}
+
+// This method resolve the path in the context dictionary
+// in case of relative path if parentProcWorkingDir == "", resolves wrt currentDir
+// otherwise wrt to parentProcWorkingDir (good when child-process change the directory)
+func resolvePathForTheFlagInContext(flagKey string, parentProcWorkingDir string, c *cli.Context) (err error) {
+	flagValue := c.String(flagKey)
+	if flagValue == "" || path.IsAbs(flagValue) {
+		return nil
+	} else if strings.HasPrefix(flagValue, "~") {
```

Comment:
all this logic is already present in getResolvedPath. dont add it at multiple places

Reply:
done.